test/serializers/associations_test.rb

```diff
@@ -261,6 +261,36 @@ def false
 
           assert_equal(expected, hash)
         end
+
+        def test_string_conditional_associations
+          serializer = Class.new(ActiveModel::Serializer) do
+            belongs_to :if_assoc_included, if: 'object.true'
+            belongs_to :if_assoc_excluded, if: 'object.false'
+            belongs_to :unless_assoc_included, unless: 'object.false'
+            belongs_to :unless_assoc_excluded, unless: 'object.true'
+          end
+
+          model = ::Model.new(true: true, false: false)
+          hash = serializable(model, serializer: serializer).serializable_hash
+          expected = { if_assoc_included: nil, unless_assoc_included: nil }
```

Comment:
took me a moment to figure out the naming is the key to the expectation here :shrug: